### PR TITLE
Extract rest client

### PR DESCRIPTION
**Pull request by wanlwanl**
Plan to add more REST APIs in Management SDK.
Plan to add check-service-health API to IServiceManager, so:
1. extract the `RestClient` from `RestHubLifetimeManager` for better reuse.
2. refactor `RestApiProvider` make it support APIs without hub name.

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Management/RestApiProvider.cs [hunk 1]
```diff
@@ -7,69 +7,71 @@ namespace Microsoft.Azure.SignalR.Management
...
-            _hubName = hubName;
-            _appName = appName;
             _restApiAccessTokenGenerator = new RestApiAccessTokenGenerator(new AccessKey(key));
-            _requestPrefix = _port == null ? $"{_baseEndpoint}/api/v1/hubs/{GetPrefixedHubName(_appName, _hubName)}" : $"{_baseEndpoint}:{_port}/api/v1/hubs/{GetPrefixedHubName(_appName, _hubName)}";
-            _audiencePrefix = $"{_baseEndpoint}/api/v1/hubs/{GetPrefixedHubName(_appName, _hubName)}";
         }
 
         private string GetPrefixedHubName(string applicationName, string hubName)
         {
             return string.IsNullOrEmpty(applicationName) ? hubName.ToLower() : $"{applicationName.ToLower()}_{hubName.ToLower()}";
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/RestApiProvider.cs:24]: take care culture for `ToLower`

File: src/Microsoft.Azure.SignalR.Management/RestClient.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            {
+                case HttpStatusCode.BadRequest:
+                {
+                    throw new AzureSignalRInvalidArgumentException(requestUri, innerException, detail);
+                }
+                case HttpStatusCode.Unauthorized:
+                {
+                    throw new AzureSignalRUnauthorizedException(requestUri, innerException);
+                }
+                case HttpStatusCode.NotFound:
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/RestClient.cs:77]: we will return `NotFound` for existence APIs.

File: src/Microsoft.Azure.SignalR.Management/RestClient.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            }
+            catch (HttpRequestException ex)
+            {
+                throw new AzureSignalRInaccessibleEndpointException(request.RequestUri.ToString(), ex);
+            }
+
+            try
+            {
+                detail = await response.Content.ReadAsStringAsync();
+                response.EnsureSuccessStatusCode();
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Management/RestClient.cs:37]: perf: should not throw the exception then catch.

File: src/Microsoft.Azure.SignalR.Management/RestApiProvider.cs (hunk 1 above)
Comment (wanlwanl) [reply src/Microsoft.Azure.SignalR.Management/RestApiProvider.cs:24]: Plan to change in another PR

**Pull request merged**